MapboxNavigationUI/RouteMapViewController.swift

```diff
@@ -240,7 +240,46 @@ extension RouteMapViewController: NavigationMapViewDelegate {
             }
         }
         
-        return CLLocation(coordinate: newCoordinate, altitude: location.altitude, horizontalAccuracy: location.horizontalAccuracy, verticalAccuracy: location.verticalAccuracy, course: location.course, speed: location.speed, timestamp: location.timestamp)
+        let defaultReturn = CLLocation(coordinate: newCoordinate, altitude: location.altitude, horizontalAccuracy: location.horizontalAccuracy, verticalAccuracy: location.verticalAccuracy, course: location.course, speed: location.speed, timestamp: location.timestamp)
+        
+        guard location.course == -1 else {
+            return defaultReturn
+        }
+        
+        guard let coords = routeController.routeProgress.route.coordinates else {
+            return defaultReturn
+        }
+        
+        guard let closest = closestCoordinate(on: coords, to: location.coordinate) else {
```

Comment:
`closestCoordinate(on:to:)` only returns `nil` if `coords` is empty. Assert that `coords` isn’t empty beforehand, then force-unwrap this result too.